src/radical/pilot/agent/launch_method/mpiexec.py

```diff
@@ -222,25 +228,36 @@ def get_launch_cmds(self, task, exec_path):
         slots = task['slots']
         sbox  = task['task_sandbox_path']
 
-        assert slots.get('ranks'), 'task.slots.ranks not defined'
+        assert slots, 'task.slots not defined'
 
         host_slots = defaultdict(int)
-        for rank in slots['ranks']:
-            host_slots[rank['node_name']] += len(rank['core_map'])
+        for slot in slots:
+            host_slots[slot['node_name']] += 1
 
         cmd_options = '-np %d ' % sum(host_slots.values())
 
         if self._use_rf:
             rankfile     = self._get_rank_file(slots, uid, sbox)
-            hosts        = set([r['node_name'] for r in slots['ranks']])
+            hosts        = set([slot['node_name'] for slot in slots])
             cmd_options += '-H %s -rf %s' % (','.join(hosts), rankfile)
 
         elif self._mpi_flavor == self.MPI_FLAVOR_PALS:
             hostfile     = self._get_host_file(slots, uid, sbox)
-            core_ids     = ':'.join([
-                str(cores[0]) + ('-%s' % cores[-1] if len(cores) > 1 else '')
-                for core_map in [rank['core_map'] for rank in slots['ranks']]
-                for cores in core_map])
+
+            tmp = list()
+            for slot in slots:
+                cores = slot['cores']
+                if len(cores) > 1:
+                    tmp.append('%s-%s' % (cores[0], cores[-1]))
+                else:
+                    tmp.append(str(cores[0]))
```

Comment:
need to add `['index']` for cores elements

Reply:
Thanks!